src/Nether.Analytics/OutputFormatters/CsvOutputFormatter.cs

```diff
@@ -60,11 +60,20 @@ public string Format(Message message)
             return builder.ToString();
         }
 
+        public string FormatWithHeaders(Message msg)
+        {
+            var sb = new StringBuilder();
+            sb.AppendLine(Header);
+            sb.Append(Format(msg));
+
+            return sb.ToString();
+        }
+
         public string Header
         {
             get
             {
-                return string.Join(Separator.ToString(), Columns) + Environment.NewLine;
```

Comment:
We can't change the behaviour of the Header property without changing the implementation of every output provider that supports headers. This implementation causes DataLakeStoreOutputProvider, FileOutputProvider and potentially more to fail. Make sure we test all before doing this change.